packages/service/src/storybook.utils.ts

```diff
@@ -0,0 +1,380 @@
+
+    try {
+        const [indexRes, storiesRes] = await Promise.all([
+            fetch(indexJsonUrl, fetchOptions),
+            fetch(storiesJsonUrl, fetchOptions),
+        ])
+
+        for (const response of [storiesRes, indexRes]) {
+            if (response.ok) {
+                const data = await response.json() as StoriesRes | IndexRes
+                return (data as StoriesRes).stories || (data as IndexRes).entries
+            }
+        }
+    } catch (err) {
+        console.error(err)
+    }
+
+    throw new Error(`Failed to fetch index data from the project. Ensure URLs are available with valid data: ${storiesJsonUrl}, ${indexJsonUrl}.`)
+}
+
+/**
+ * Get arg value from the process.argv
+ */
+export function getArgvValue(argName: string, parseFunc: (value: string) => any): any {
+    const argWithEqual = argName + '='
+    const argv = process.argv
+
+    for (let i = 0; i < argv.length; i++) {
+        if (argv[i] === argName && i + 1 < argv.length) {
+            return parseFunc(argv[i + 1])
+        } else if (argv[i].startsWith(argWithEqual)) {
+            const value = argv[i].slice(argWithEqual.length)
+            return parseFunc(value)
+        }
+    }
+
+    return undefined
+}
+
+/**
+ * Creates a it function for the test file
+ * @TODO: improve this
+ */
+export function itFunction({ clip, clipSelector, folders: { baselineFolder }, framework, skipStories, storyData, storybookUrl }: CreateItContent) {
+    const { id } = storyData
+    const screenshotType = clip ? 'n element' : ' viewport'
+    const DEFAULT_IT_TEXT = 'it'
+    let itText = DEFAULT_IT_TEXT
+
+    if (isJasmineFramework(framework)) {
+        itText = 'xit'
+    } else if (isMochaFramework(framework)) {
+        itText = 'it.skip'
+    }
+
+    if (Array.isArray(skipStories)) {
+        itText = skipStories.includes(id) ? itText : DEFAULT_IT_TEXT
+    } else if (skipStories instanceof RegExp) {
+        itText = skipStories.test(id) ? itText : DEFAULT_IT_TEXT
+    }
+
+    // Setup the folder structure
+    const { category, component } = extractCategoryAndComponent(id)
+    const methodOptions = {
+        baselineFolder: join(baselineFolder, `./${category}/${component}/`),
+    }
+
+    const it = `
+    ${itText}(\`should take a${screenshotType} screenshot of ${id}\`, async () => {
+        await browser.url(\`${storybookUrl}iframe.html?id=${id}\`);
+        await $('#storybook-root').waitForDisplayed();
+
+        const startTime = performance.now();
+        ${clip
+        ? `await expect($('${clipSelector}')).toMatchElementSnapshot('${id}-element', ${JSON.stringify(methodOptions)})`
+        : `await expect(browser).toMatchScreenSnapshot('${id}', ${JSON.stringify(methodOptions)})`}
+        // await $('${clipSelector}').saveScreenshot('${id}.png');
+        const endTime = performance.now();
+        const timeTaken = endTime - startTime;
+        console.log('Visual snapshot of \`${id}\` took: ', Math.round(timeTaken), ' ms')
```

Comment:
Should this be a wdio log so people would be able to have it written into a wdio log file?
```suggestion
        console.log('Visual snapshot of \`${id}\` took: ', Math.round(timeTaken), ' ms')
```

Reply:
No, this needs to be removed 😅 , will push a fix later today